selectors/views/custom_status.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+import {Preferences} from 'mattermost-redux/constants';
+
+import {createSelector} from 'reselect';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
+
+import {GlobalState} from 'types/store';
+
+export function makeGetCustomStatus() {
+    return memoizeResult((state: GlobalState, userID?: string) => {
+        const user = userID ? getUser(state, userID) : getCurrentUser(state);
+        const userProps = user.props || {};
+        return userProps.customStatus && JSON.parse(userProps.customStatus);
+    });
```

Comment:
This doesn't quite work since `state` will be changing constantly. Instead, you can use `createSelector` and vary the user that's passed into its final function based on whether or not the user ID is provided
```suggestion
    return createSelector(
        (state: GlobalState, userID?: string) => userID ? getUser(state, userID) : getCurrentUser(state),
        (user) => {
            const userProps = user.props || {};
            return userProps.customStatus && JSON.parse(userProps.customStatus);
        },
    );
```
This will make it so that the returned value is only changed whenever `user` changes rather than whenever anything in `state` changes. You could go a bit further and make it only pass the props into that final function, but that's probably not necessary.